### ISSUE TITLE
`onKeyDown` not fired for editable elements nested inside non-editables

### ISSUE THREAD

**Issue by AlbertHilb** [ianstormtaylor/slate#468]
In my app I use container blocks rendered as

```
<section className="definition" contentEditable={false}>
  <header>Definition.</header>
  <div className="content" contentEditable={true}>
    {props.children}
  </div>
</section>
```
Unfortunately editing their children does not trigger the 'onKeyDown' event handler.
This happens because the `content` component use the test

```
(
  (target.isContentEditable) &&
  (target === element || target.closest('[contenteditable]') == element)
)
```
to decide if an 'onKeyDown' handler must be called.
According to the comment in the source code, the test is intended to:
 1. prevent 'onKeyDown' handlers being called for non-editable elements 
     ```target.isContentEditable```
 1. ignore events coming from nested slate editors
     ```target === element || target.closest('[contenteditable]') == element```

In the second check it is assumed that the only ancestors of an element with `contenteditable` attribute are slate `content` component instances.
That assumption, of course, is wrong in my case and the test fails even if it should not.

My proposal to solve the problem is to add an attribute to slate `content`elements, let's say `data-slate-root`, to uniquely identify them, and then to rewrite the test as

```
(
  (target.isContentEditable) &&
  (target === element || target.closest('[data-slate-root]') === element)
)
 ```

If you agree I will make a PR.

**Comment by ianstormtaylor** [comment]
Hey @AlbertHilb, cool use case, I totally agree with your conclusion! I'd be happy to merge that PR. Thank you!

**Comment by tyler-johnson** [comment]
Just noticed this issue and seeing as I submitted #443 which changed the functionality to this, I feel like I should weigh in.

Perhaps the issue here is that we are treating all the events the same? For my nested use case, the issue wasn't that all events were bubbling, but rather that the React `onInput` event was bubbling. This was causing both Slate instances to interpret the changes resulting in undesired behavior. Maybe only the `onInput` event should use `[content-editable]` and the other events use `[data-slate-content]`?

**Comment by tyler-johnson** [comment]
Actually ignore my "solution" as I see it still runs into the same issue of the nested content-editables. Still I could see there being nested content-editables that should not be a part of the Slate instance and therefore should not have their `onInput` bubble. #470 would not solve that usecase.

**Comment by AlbertHilb** [comment]
Hi @tyler-johnson. :blush: 
What exactly is the use-case not fixed by #470?
Thanks!

**Comment by tyler-johnson** [comment]
Well so I was thinking this could be an issue:

```html
<div content-editable="false">
  <div content-editable="true">
    <!-- not a slate editor -->
  </div>
</div>
```

However, today I realize that all I would need to do is manually prevent the event bubbling for that inner content-editable. Haha I was a bit sick yesterday so please ignore my ramblings 😆
